packages/headless/src/controllers/triggers/headless-execute-trigger.ts

```diff
@@ -0,0 +1,90 @@
+import {TriggerSection} from '../../state/state-sections';
+import {triggers} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {logTriggerExecute} from '../../features/triggers/trigger-analytics-actions';
+import {SearchEngine} from '../../app/search-engine/search-engine';
+
+/**
+ * The `ExecuteTrigger` controller handles execute trigger actions.
+ */
+export interface ExecuteTrigger extends Controller {
+  /**
+   * the state of the `ExecuteTrigger` controller.
+   */
+  state: ExecuteTriggerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `ExecuteTrigger` controller.
+ */
+export interface ExecuteTriggerState {
+  /**
+   * The name of the function to be executed.
+   */
+  name: string;
+
+  /**
+   * The parameters of the function to be executed.
+   */
+  params: {}[];
+}
+
+/**
+ * Creates a `ExecuteTrigger` controller instance.
```

Comment:
I think it would be useful to expand the description here, or link to some documentation, or maybe both. e.g.

```
Creates a `ExecuteTrigger` controller instance. An execute trigger is configured in the Administration console, and used to execute a function in the browser when a certain condition is met.
```

Doc: https://docs.coveo.com/en/1458/tune-relevance/trigger-query-pipeline-feature

cc: @dlutzCoveo

Reply:
We could expand the description on all the other trigger controllers as well.